src/lib/caching/stores/GuildChannelInviteStore.ts

```diff
@@ -0,0 +1,78 @@
+import type { Client } from '../../client/Client';
+import { ProxyCache } from '@klasa/cache';
+import type { Invite } from '../structures/Invite';
+import type { GuildChannel } from '../structures/channels/GuildChannel';
+import { RequestOptions, Routes } from '@klasa/rest';
+import type { APIInviteData } from '@klasa/dapi-types';
+import type { Channel } from '../structures/channels/Channel';
+import type { TextChannel } from '../structures/channels/TextChannel';
+/**
+ * The store for {@link Invite guild invites} the channel has.
+ * @since 0.0.3
+ */
+export class GuildChannelInviteStore extends ProxyCache<string, Invite> {
+
+	/**
+	 * The {@link Client client} this store belongs to.
+	 * @since 0.0.3
+	 */
+	public readonly client: Client;
+
+	/**
+	 * The {@link Channel channel} this store belongs to.
+	 * @since 0.0.3
+	 */
+	public readonly channel: Channel;
+
+	/**
+	 * Builds the store.
+	 * @since 0.0.3
+	 * @param channel The {@link GuildChannel guild channel} this store belongs to.
+	 */
+	public constructor(channel: GuildChannel, keys: string[]) {
+		super(channel.invites, keys);
+		this.client = channel.client;
+		this.channel = channel;
+	}
+
+	/**
+	 * Creates an invite to the channel.
+	 * @since 0.0.3
+	 * @param requestOptions The additional request options.
+	 * @see https://discord.com/developers/docs/resources/channel#create-channel-invite
+	 */
+	public async create(requestOptions: RequestOptions = {}): Promise<this> {
+		const entry = await this.client.api.post(Routes.channelInvites(this.channel.id), requestOptions) as APIInviteData;
+		this.set(entry.code);
```

Comment:
You should also call the main sources _add so that if caching is enabled, it can be cached properly.